macq/generate/pddl/vanilla_sampling.py

```diff
@@ -112,6 +112,8 @@ def generate_single_trace(self=self, plan_len=plan_len):
 
             if not plan_len:
                 plan_len = self.plan_len
+            if callable(plan_len):
```

Comment:
Following from the above comment, if a callable `plan_len` is called right away, why not call it in the initialization of the `FDRandomWalkSampling`?

Reply:
The key thing is that plan length is _repeatedly_ called -- it's sampling the length each time a new trace is called for.

Reply:
oh I see, `_plan_len` is non-deterministic that's the piece I was missing